pandas/_libs/tslibs/timezones.pyx

```diff
@@ -76,6 +76,15 @@ cpdef inline object get_timezone(object tz):
                 zone = tz.zone
                 if zone is None:
                     return tz
+
+                # UTC and Etc/GMT or Etc/GMT+0 are the same timezones
```

Comment:
this is WAY too specific

what are you trying to do here? we don’t compare time zones like this

Reply:
not to mention likely non perform any

Reply:
if you find a case that is failing make a new issue and just fail it here

this PR aid already a lot

Reply:
Moreover, string timezones should pass though `maybe_get_tz`, not here.